### ISSUE TITLE
azure_rm_subscription_info tagging bug

### ISSUE THREAD

**Issue by HichamMourad** [ansible/ansible_collections_azure#807]
##### SUMMARY
getting the tags deos NOT work as described in the documentation.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
azure_rm_subscription_info
  tags:
      - foo:bar

##### ANSIBLE VERSION

```paste below
ansible [core 2.12.1]
  config file = None
  configured module search path = ['/Users/hmourad/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/hmourad/Documents/GitHub/hm-py-venv/lib/python3.8/site-packages/ansible
  ansible collection location = /Users/hmourad/.ansible/collections:/usr/share/ansible/collections
  executable location = /Users/hmourad/Documents/GitHub/hm-py-venv/bin/ansible
  python version = 3.8.5 (v3.8.5:580fbb018f, Jul 20 2020, 12:11:27) [Clang 6.0 (clang-600.0.57)]
  jinja version = 3.0.3
  libyaml = True
```

##### COLLECTION VERSION

```paste below
1.12.0
```

##### CONFIGURATION

```paste below
END
```

##### OS / ENVIRONMENT

OSX Monterey

##### STEPS TO REPRODUCE

the below task generates the following error::::

TASK [open-env-azure-add-user-to-subscription : Get facts for subscriptions containing tags AVAILABLE.] ******************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Elements value for option 'tags' is of type <class 'str'> and we were unable to convert to dict: dictionary requested, could not parse JSON or key=value"}

```yaml

- name: Get facts for subscriptions containing tags AVAILABLE.
  azure.azcollection.azure_rm_subscription_info:
    auth_source: cli
    tags:
      - AVAILABLE:TRUE
  register: available_subs

```

##### ACTUAL RESULTS

```paste below

TASK [open-env-azure-add-user-to-subscription : Get facts for subscriptions containing tags AVAILABLE.] ******************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Elements value for option 'tags' is of type <class 'str'> and we were unable to convert to dict: dictionary requested, could not parse JSON or key=value"}


```

**Comment by Fred-sun** [comment]
@HichamMourad Try the playbook below! Thank you very much!
```e.g.
- name: Get facts for subscriptions containing tags AVAILABLE.
  azure.azcollection.azure_rm_subscription_info:
    auth_source: cli
    tags:
      - AVAILABLE: TRUE ------------------This should be a dictionary（tags=dict(type='list', elements='dict')）
  register: available_subs

**Comment by Fred-sun** [comment]
fixes by #819
